### PR TITLE
Common configuration for test resources

### LINKED ISSUE #43: Add common configuration for test resources
Either pattern in `[general]` config or common setting for all plugins. With latter approach we will still be able to treat, e.g., pair of `*Test` and `*Expected` files as a single test.

And this API can be used in preprocessor in save-cloud to dsicover all tests for any generic cofnig

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Changed PluginConfig, added `discoverTestFiles` method to Plugin
* Updated fix-plugin and warn-plugin
* Fixed tests

Closes #43

### CODE REVIEW

File: save-core/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt [hunk 1]
```diff
@@ -16,4 +18,12 @@ interface Plugin {
...
      */
     fun execute(saveProperties: SaveProperties, testConfig: TestConfig): Sequence<TestResult>
+
+    /**
+     * Discover groups of resource files which will be used to run tests.
+     *
+     * @param root root [Path], from where discovering should be started
+     * @return a sequence of files, grouped by test
+     */
+    fun discoverTestFiles(root: Path): Sequence<List<Path>>
```
Comment (kgevorkyan) [inline save-core/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:28]: It seems, that return value actually is `Sequence<Pair<Path>>`, right?

May be we can explicitly specify this?

Comment (petertrr) [reply save-core/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:28]: No, `Pair` and `List` are not related, so they couldn't be cast to each other. We have a list with single element for `warn-plugin` and a list of two elements for `fix-plugin`, so to keep the API more general I've put a list here

Comment (kgevorkyan) [reply save-core/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:28]: Ok, just saw something like `.map { it.first() to it.last() }`, so I thought, that we have the `Pair`. Also previously we had `discoverFilePairs`

File: save-core/src/commonMain/kotlin/org/cqfn/save/core/config/TestConfig.kt [hunk 2]
```diff
@@ -14,13 +15,25 @@ data class TestConfig(
...
      */
     val childConfigs: MutableList<TestConfig> = mutableListOf()
 
+    /**
+     * Directory containing [location] of this config
+     */
+    val directory: Path = location.parent!!
+
+    init {
+        require(!fs.metadata(location).isDirectory) {
```
[the remark this answers is not in the document]
Comment (petertrr) [reply save-core/src/commonMain/kotlin/org/cqfn/save/core/config/TestConfig.kt:32]: I think we can, I will change

**Review by kgevorkyan — APPROVED** [review]

**Review by akuleshov7 — APPROVED** [review]
lgtm

**Pull request merged**